R/slack.R

```diff
@@ -66,8 +56,76 @@ slack_data <- function(dat, remote_name, report_url, remote_is_primary) {
            url = report_url)))))
 }
 
+teams_data <- function(dat, remote_name, report_url, remote_is_primary) {
+  content <- prepare_content(dat, remote_name, report_url, "[{label}]({url})")
+  facts <- list(list(name = "id:", value = content$id))
+  if (!is.null(content$git)) {
+    facts <- c(facts, list(list(name = "git:", value = content$git)))
+  }
+  ## Using same colours as slack default for "good" and "warning" messages
+  col <- if (remote_is_primary) "2EB886" else "DAA038"
+
+  list(
+    "@type" = "MessageCard",
+    "@context" = "http://schema.org/extensions",
+    summary = content$fallback,
+    themeColor = col,
+    sections = list(
+      list(
+        activityTitle = content$title,
+        activityText = content$text,
+        ## TODO: Choose another image from somewhere?
+        activityImage = paste0("https://cdn.pixabay.com/photo/2017/",
+                               "06/10/07/18/list-2389219_960_720.png")
+      ),
+      list(
+        facts = facts
+      )
+    ),
+    potentialAction = list(
+      list(
+        "@type" = "OpenUri",
+        name = "View report",
+        targets = list(
+          list(
+            os = "default",
+            uri = report_url
+          )
+        )
+      )
+    )
+  )
+}
+
+prepare_content <- function(dat, remote_name, report_url, link_format) {
+  id <- dat$meta$id
+  name <- dat$meta$name
+  elapsed <- format(as.difftime(dat$meta$elapsed, units = "secs"), digits = 2)
+  git <- dat$git
+  if (!is.null(git)) {
+    branch <- git$branch %||% "(detached)"
+    sha <- git$sha_short
+    if (!is.null(git$github_url)) {
+      url <- sprintf("%s/tree/%s", git$github_url, sha)
+      sha <- glue::glue(link_format, url = url, label = sha)
```

Comment:
Removed that dependency and have moved docopt and crayon to suggests section too

Reply:
Sorry, wasn't meaning for you to have to do that, and it will require some fiddling to wrappers to make sure that the functionality works without them present - including some faff with docker perhaps. Could you move crayon and docopt back and we'll take care of that on VIMC-3664